subsys/bluetooth/host/att.c

```diff
@@ -378,11 +378,12 @@ static u8_t att_handle_rsp(struct bt_att *att, void *pdu, u16_t len, u8_t err)
 	att_req_destroy(att->req);
 	att->req = NULL;
 
-	func(att->chan.chan.conn, err, pdu, len, params);
-
 process:
 	/* Process pending requests */
 	att_process(att);
```

Comment:
`att_process` will not perform any blocking procedures AFAICT so putting the callback after `att_process` seems acceptable to me

We discovered this issue after pulling the changes for ATT request pool.
Before that the code in question looked like this:
```

	func(att->chan.chan.conn, err, pdu, len, att->req);

	/* Don't destroy if callback had reused the request */
	if (!att->req->func) {
		att_req_destroy(att->req);
	}

	att->req = NULL;

process:
	/* Process pending requests */
	att_process(att);
```
In this case `att->req` is not NULL during the execution of the callback. So any requests queued by the application would not be put into this slot.